demo/run_demo.py

```diff
@@ -128,6 +128,7 @@ def main():
       "final_product/foo.tar.gz",
       "final_product/package.link",
       "final_product/write-code.link",
+      "final_product/tar.link",
```

Comment:
We didn't dump inspections before, ergo there was no `tar.link`.  `verifylib. run_all_inspections` now calls `runlib.in_toto_run` which also dumps the metadata file, which we wanted to add for inspections (auditing reasons) anyway. Therefor, I also remove this file when clearing the demo final_product.